using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md

```diff
@@ -0,0 +1,520 @@
+    @Override    
+    public void call(String id) {        
+         //Handle the ID returned here    
+    }
+});
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### getExperienceCloudId
+
+This API retrieves the ECID that was generated when the app was initially launched.
+
+This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall. The values are returned via the callback.
+
+**Syntax**
+
+```swift
+@objc(getExperienceCloudId:)
+static func getExperienceCloudId(completion: @escaping (String?, Error?) -> Void)
+```
+
+* _completion_ is invoked after the ECID is available.  The default timeout is 1000ms.
+
+**Examples**
+
+**Objective-C**
+
+```objectivec
+[AEPEdgeIdentity getExperienceCloudId:^(NSString * _Nullable ecid, NSError * error) {   
+    // handle the retrieved ID here    
+}];
+```
+
+**Swift**
+
+```swift
+Identity.getExperienceCloudId { (ecid, error) in
+  if let error = error {
+    // handle error here
+  } else {
+    // handle the retrieved ID here
+  }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+## getIdentities
+{% tabs %}
+{% tab title="Android" %}
+### getIdentities
+
+Get all identities in the Edge Identity extension, including customer identifiers which were previously added.
+
+When [AdobeCallbackWithError](../mobile-core/mobile-core-api-reference.md#public-classes) is provided, and you are fetching the identities from the Mobile SDK, the timeout value is 500ms. If the operation times out or an unexpected error occurs, the `fail` method is called with the appropriate [AdobeError](../mobile-core/mobile-core-api-reference.md#public-classes).
+
+**Java**
+
+**Syntax**
+
+```java
+public static void getIdentities(final AdobeCallback<IdentityMap> callback);
+```
+
+* _call_ is invoked after the identities are available. The return format is an instance of [IdentityMap](#identitymap)
+
+**Example**
+
+```java
+Identity.getIdentities(new AdobeCallback<IdentityMap>() {    
+    @Override    
+    public void call(IdentityMap identityMap) {        
+         //Handle the IdentityMap returned here    
+    }
+});
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### getIdentities
+
+Get all identities in the Edge Identity extension, including customer identifiers which were previously added.
+
+**Syntax**
+
+```swift
+@objc(getIdentities:)
+static func getIdentities(completion: @escaping (IdentityMap?, Error?) -> Void)
+```
+
+* _completion_ is invoked after the identities are available.  The default timeout is 1000ms. The return format is an instance of [IdentityMap](#identitymap)
+
+**Examples**
+
+**Objective-C**
+
+```objectivec
+[AEPEdgeIdentity getIdentities:^(AEPIdentityMap *map, NSError * error) {   
+    // handle the retrieved identities here    
+}];
+```
+
+**Swift**
+
+```swift
+Identity.getIdentities { (identityMap, error) in
+  if let error = error {
+    // handle error here
+  } else {
```

Comment:
Same question here about unwrapping the identity map as it's marked as optional.